python/pyproject.toml

```
@@ -288,3 +281,47 @@ ignore_missing_imports = true
 
 [tool.coverage.run]
 source = ['pyiceberg/']
+
+[tool.ruff]
+src = ['pyiceberg/', 'tests/']
+# Enable the pycodestyle (`E`) and Pyflakes (`F`) rules by default.
+# Unlike Flake8, Ruff doesn't enable pycodestyle warnings (`W`) or
+# McCabe complexity (`C901`) by default.
+select = ["E", "F"]
```

When we remove certain other checkers, we have to make sure that they are enabled:
```suggestion
select = ["E", "F", "W", "I", "UP"]
```

Another one is pydocstyle (`D`), but then we have to add some ignores as well, since we're not there yet: https://github.com/apache/iceberg/issues/7776